layer/tests/layer_tests_main.cpp

```diff
@@ -0,0 +1,286 @@
+        AAsset *asset = AAssetManager_open(assetManager, filename, AASSET_MODE_BUFFER);
+        if (asset != nullptr) {
+          const void* file_buffer = AAsset_getBuffer(asset);
+          off_t file_length = AAsset_getLength(asset);
+          std::string dst_filename = JSON_TEST_FILES_PATH;
+          dst_filename += filename;
+          if (int(dst_filename.back() == 0xa) ) {
+            dst_filename.pop_back();
+          }
+          FILE* outfile = fopen(dst_filename.c_str(), "w");
+          if (outfile == nullptr) {
+              print_android("Vulkan Profiles Layer Test: failed to open destination file");
+          }
+          fwrite(file_buffer, file_length, 1, outfile);
+          fclose(outfile);
+          AAsset_close(asset);
+        } else {
+          print_android("Vulkan Profiles Layer Test: Failed to open asset file");
+        }
+      }
+      AAssetDir_close(assetDir);
+    } else {
+      print_android("Vulkan Profiles Layer Test: Failed to open root directory");
+    }
+  } else {
+    print_android("Vulkan Profiles Layer Test: Asset Manager was invalid");
+  }
+}
+
+// Convert Intents to argv
+std::vector<std::string> get_args(android_app &app, const char *intent_extra_data_key) {
+    std::vector<std::string> args;
+    JavaVM &vm = *app.activity->vm;
+    JNIEnv *p_env;
+    if (vm.AttachCurrentThread(&p_env, nullptr) != JNI_OK) return args;
+
+    JNIEnv &env = *p_env;
+    jobject activity = app.activity->clazz;
+    jmethodID get_intent_method = env.GetMethodID(env.GetObjectClass(activity), "getIntent", "()Landroid/content/Intent;");
+    jobject intent = env.CallObjectMethod(activity, get_intent_method);
+    jmethodID get_string_extra_method =
+        env.GetMethodID(env.GetObjectClass(intent), "getStringExtra", "(Ljava/lang/String;)Ljava/lang/String;");
+    jvalue get_string_extra_args;
+    get_string_extra_args.l = env.NewStringUTF(intent_extra_data_key);
+    jstring extra_str = static_cast<jstring>(env.CallObjectMethodA(intent, get_string_extra_method, &get_string_extra_args));
+
+    std::string args_str;
+    if (extra_str) {
+        const char *extra_utf = env.GetStringUTFChars(extra_str, nullptr);
+        args_str = extra_utf;
+        env.ReleaseStringUTFChars(extra_str, extra_utf);
+        env.DeleteLocalRef(extra_str);
+    }
+
+    env.DeleteLocalRef(get_string_extra_args.l);
+    env.DeleteLocalRef(intent);
+    vm.DetachCurrentThread();
+
+    // split args_str
+    std::stringstream ss(args_str);
+    std::string arg;
+    while (std::getline(ss, arg, ' ')) {
+        if (!arg.empty()) args.push_back(arg);
+    }
+
+    return args;
+}
+
+void addFullTestCommentIfPresent(const ::testing::TestInfo &test_info, std::string &error_message) {
+    const char *const type_param = test_info.type_param();
+    const char *const value_param = test_info.value_param();
+
+    if (type_param != NULL || value_param != NULL) {
```

Comment:
Can we use `nullptr` instead of `NULL` and elsewhere too?

Reply:
Absolutely, will do.